hudi-common/src/main/java/org/apache/hudi/common/model/HoodieRecord.java

```diff
@@ -311,6 +311,17 @@ public void checkState() {
 
   protected abstract T readRecordPayload(Kryo kryo, Input input);
 
+  /**
+   * Clears the new currentLocation of the record. 
+   *
+   * This is required in the delete path so that Index can track that this record was deleted.
+   */
+  public HoodieRecord clearNewLocation() {
```

Comment:
why return HoodieRecord. can't we return void ?

Reply:
Fixed. Code was copied from setNewLocation which also needs fix.